include/linker/devicetree_regions.h

```diff
@@ -37,25 +37,25 @@
 		    (DT_PROP(node_id, zephyr_linker_region)),	     \
 		    (DT_NODE_PATH(node_id)))
 
-/* Declare a memory region */
-#define _REGION_DECLARE(name, attr, node) name(attr) : \
-	ORIGIN = DT_REG_ADDR(node),		       \
+/**
+ * @brief Declare a memory region
+ *
+ * @param node devicetree node
+ * @param attr region attributes
+ */
+#define _REGION_DECLARE(node, attr)	    \
+	DT_NODE_LINKER_REGION(node)(attr) : \
+	ORIGIN = DT_REG_ADDR(node),	    \
 	LENGTH = DT_REG_SIZE(node)
 
 /**
  * @brief Generate a linker memory region from a devicetree node
  *
- * If @p node_id refers to a node with status "okay", then this declares
- * a linker memory region named @p name with attributes from @p attr.
- *
- * Otherwise, it doesn't expand to anything.
- *
- * @param name name of the generated memory region
+ * @param node devicetree node with a \<reg\> property defining region location
+ *             and size.
  * @param attr region attributes to use (rx, rw, ...)
- * @param node_id devicetree node identifier with a \<reg\> property
- *                defining region location and size.
  */
-#define LINKER_DT_REGION_FROM_NODE(name, attr, node_id)		\
-	COND_CODE_1(DT_NODE_HAS_STATUS(node_id, okay),		\
-		    (_REGION_DECLARE(name, attr, node_id)),	\
+#define LINKER_DT_REGION_FROM_NODE(node, attr) \
+	COND_CODE_1(DT_NODE_HAS_STATUS(node, okay), \
+		    (_REGION_DECLARE(node, attr)),  \
```

Comment:
Any reason why you reverted the macro parameter changes in here? It seems unrelated to the purpose of the commit.

I had changed to `node_id` on purpose, because that makes it more consistent with other APIs that take them as parameters.

Reply:
I missed it in the rebase, sorry

Reply:
> I missed it in the rebase, sorry

Something is still up with the latest rebase as this is still using the DT_ prefix and this is still getting refactored. Please double check.